### PR TITLE
Reconfigure cmake preset

### PR DESCRIPTION
**Pull request by TonyXiang8787**
Let Cmake presets to choose the compiler `gcc` and `clang` automatically.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -42,16 +42,17 @@ endif()
...
 add_subdirectory("power_grid_model_c")
 
-# get tests
-if(${PGM_ENABLE_TESTING})
+# dev build
+if(${PGM_ENABLE_DEV_BUILD})
   include(CTest)
   enable_testing()
-
+# get tests
```
Comment (mgovers) [inline CMakeLists.txt:49]: nitpick format
```suggestion
  # get tests
```

File: CMakeLists.txt [hunk 2]
```diff
@@ -42,16 +42,17 @@ endif()
...
   add_subdirectory("tests")
+  # get c api example
+add_subdirectory("power_grid_model_c_example")
```
Comment (mgovers) [inline CMakeLists.txt:52]: nitpick format
```suggestion
  # get c api example
  add_subdirectory("power_grid_model_c_example")
```

File: CMakePresets.json [hunk 3]
```diff
@@ -125,7 +127,9 @@
       "name": "clang-tidy-unix-base",
       "cacheVariables": {
         "CMAKE_C_CLANG_TIDY": "$env{CLANG_TIDY}",
-        "CMAKE_CXX_CLANG_TIDY": "$env{CLANG_TIDY}"
+        "CMAKE_CXX_CLANG_TIDY": "$env{CLANG_TIDY}",
```
Comment (mgovers) [inline CMakePresets.json:130]: it's probably good to also do the `clang-tidy` the same way you do `clang`/`clang++` using symlink

**Review by mgovers — DISMISSED** [review]

File: docs/advanced_documentation/build-guide.md [hunk 4]
```diff
@@ -128,17 +128,17 @@ In principle, you can use any C++ IDE with cmake and ninja support to develop th
 the bare CMake CLI to set up the project. For ease of use, several presets are available (CMake 3.23+). Supported presets
 for your development platform can be listed using `cmake --list-presets`.
 
+In Linux/macOS, the presets will use command `clang++`/`clang` or `gcc`/`g++` to find the relevant `gcc` or `clang` compiler. It is the developer's reponsiblity to proper define some symbolic link (which should be findable through `PATH` environment) of your actual `clang` or `gcc` compiler in your system. If you want to build with `clang-tidy`, you also need to define symbolic link of `clang-tidy` to point to the actual `clang-tidy` program of your system.
```
Comment (Jerry-Jinfeng-Guo) [inline docs/advanced_documentation/build-guide.md:131]:
```suggestion
On Linux/macOS, the presets will use command `clang++`/`clang` or `gcc`/`g++` to find the relevant `gcc` or `clang` compiler. It is the developer's reponsiblity to properly define symbolic links (which should be discoverable through `PATH` environment variable) of `clang` or `gcc` compiler in your system. If you want to build with `clang-tidy`, you also need to define symbolic link of `clang-tidy` to point to the actual `clang-tidy` executable of your system.
```

File: docs/advanced_documentation/build-guide.md [hunk 5]
```diff
@@ -128,17 +128,17 @@ In principle, you can use any C++ IDE with cmake and ninja support to develop th
...
+
+Similar also applies for Windows, the presets will use command `cl.exe` or `clang-cl.exe` to find the compiler. Developer needs to make sure the they are findable in the `PATH`. For x64 Windows native development using MSVC or Clang CL, use the `x64 Native Command Prompt`, which uses `vcvarsall.bat` to set the appropriate build environment.
```
Comment (Jerry-Jinfeng-Guo) [inline docs/advanced_documentation/build-guide.md:133]:
```suggestion
Similar also applies to Windows: the presets will use command `cl.exe` or `clang-cl.exe` to find the compiler. Developer needs to make sure the they are discoverable in `PATH`. For x64 Windows native development using MSVC or Clang CL, please use the `x64 Native Command Prompt`, which uses `vcvarsall.bat` to set up the appropriate build environment.
```

**Review by Jerry-Jinfeng-Guo — COMMENTED** [review]
Since it is still one check away from merge

**Comment by TonyXiang8787** [comment]
@mgovers can you have a look at the package test error?

**Review by mgovers — APPROVED** [review]

**Pull request merged**